### DISCUSSION TITLE
Beam center and image range

### DISCUSSION THREAD

**Discussion by Javalde91** [dials/dials#2628] [General]
Hello everyone,

As a newie using DIALS for microED, I find myself with a couple of queries that could benefit from your expertise.

Firstly:
Is there a method within DIALS to annotate or draw lines on the images for the purpose of identifying the beam center? I'm particularly interested in a functionality similar to what's illustrated in the Biotin tutorial.

Secondly:
Upon analysis, I've identified images with an Rmerge exceeding 1. I'm considering excluding them to enhance the data quality for cosym. However, these problematic images are not confined to the beginning or end of the datasets—they are dispersed in the middle (two or three in each dataset). Is there a recommended approach for using dials.import with three distinct image ranges to accommodate this scenario?

Thank you!

**Comment by dagewa** [comment]
Hi,

1. Sorry, there isn't a facility for this in the image viewer. The biotin tutorial is a bit of a cheat, because those lines have just been added on to a screenshot. What you can do is identify some Friedel pairs by eye (the Stack function can be useful to include a few images) then use the mouse cursor and read out the fast, slow coordinates of each spot in the pair. There is not much development in the `dials.image_viewer` these days, but I could think about adding a line tool if I have time. Also, it is worth noting that we have some ongoing effort to develop automated beam centre finding algorithms that work well for ED data, so eventually this step might become trivial

2. I would not normally exclude the images at the `dials.import` stage. Normally I would exclude images at the point of running `dials.scale`. I see that you want to exclude them prior to `dials.cosym` though, which is typically run on the integrated files. Well, in that case you could use the parameter `exclude_images=0:100:110` in `dials.integrate`, which excludes images 100 to 110 inclusive from the first experiment (numbered `0`). If you are processing a single crystal then that experiment will be number `0`.

I do however have doubts that excluding the frames will matter for the cosym analysis. Typically I would exclude them in scaling, and normally I don't think about manually working out which frames to exclude based on Rmerge. `dials.scale` can automatically exclude groups of images according to the ΔCC1/2 metric. You would do that like this, for example:

```
dials.scale integrated_1.{expt,refl} integrated_2.{expt,refl} ...\
    filtering.method=deltacchalf\
    deltacchalf.mode=image_group\
    deltacchalf.group_size=5\
```

This will consider each block of 5 images and determine whether that block positively contributes to the overall CC1/2 or not. This is more robust than using Rmerge. Even if a few frames look like they have bad Rmerge, you may find that they still contribute to CC1/2, so it is worth keeping them. There are a few other parameters you can play with too, which you can look at by running:

```
dials.scale -hv
```
then looking for the scope named `deltacchalf`. Sometimes I play with the `stdcutoff` to exclude more or fewer images.

Hope this helps.

**Comment by Javalde91** [comment]
Thank you David, for your response. 
I actually use a transparent rule to find the beam center but I felt a bit outdated using it haha.

I think what you mention for dials.scale would be more useful for me since I have different images that might affect my results and they are not in a specific range. For instance, I have one at 10, another one at 160, and they are not consecutive. 
I'll give it a try with your examples! 

Thanks
